generic.go

```diff
@@ -161,13 +195,48 @@ func (g *Generic) Run(ctx context.Context) {
 		stage.Start(g.worker)
 	}
 
+	// Stage: start servers
+	stage = stgr.NextStage()
+	ctx, cancel := context.WithCancel(ctx)
+
+	for _, srv := range g.Servers {
+		stage.StartWithContext(func(metricsCtx context.Context) {
+			defer cancel() // if srv fails to start it signals the whole program that it should shut down
+			err := srv.Server.Run(metricsCtx)
```

Comment:
We should change the Generic's `Run()` to return `error` to properly fail the program. Otherwise it will log this and still exist with 0.